src/ConfigureMentions.php

```diff
@@ -42,14 +43,22 @@ private function configureUserMentions(Configurator $config)
         $tagName = 'USERMENTION';
 
         $tag = $config->tags->add($tagName);
-        $tag->attributes->add('username');
         $tag->attributes->add('displayname');
         $tag->attributes->add('id')->filterChain->append('#uint');
 
-        $tag->template = '<a href="{$PROFILE_URL}{@username}" class="UserMention">@<xsl:value-of select="@displayname"/></a>';
+        $tag->template = '
+            <xsl:choose>
+                <xsl:when test="@deleted != 1">
+                    <a href="{$PROFILE_URL}{@slug}" class="UserMention">@<xsl:value-of select="@displayname"/></a>
+                </xsl:when>
+                <xsl:otherwise>
+                    <span class="UserMention UserMention--deleted">@<xsl:value-of select="@displayname"/></span>
+                </xsl:otherwise>
+            </xsl:choose>';
         $tag->filterChain->prepend([static::class, 'addUserId'])
             ->setJS('function(tag) { return flarum.extensions["flarum-mentions"].filterUserMentions(tag); }');
 
+        $config->Preg->match('/\B@"(?<displayname>((?!"#[a-z]{0,3}[0-9]+).)+)"#(?<id>[0-9]+)\b/', $tagName);
         $config->Preg->match('/\B@(?<username>[a-z0-9_-]+)(?!#)/i', $tagName);
```

Comment:
Should this one always be supported?

Reply:
At first I implemented it so that it would not be the case: https://github.com/flarum/mentions/pull/65/commits/c12532c1c6d45f7001d0abdbf6d66dedde0febc0

But then I had to change that up, because of caching (configuration gets cached), changing the setting doesn't take effect until the cache is manually cleared.

Reply:
so it technically isn't always supported because of the check we run in `addUserId` and in the corresponding JS function.